Ipkhchan/ageimals-leaderboard
# implement posting results from poll

What this PR implments:

- when /postwinners [correct answer] is run, server looks up past messages to find the Poll message, parses the users from it and calls postLosers and postWinners to update the database.

To Test:
1. postwinners handler requires ACCESS_TOKEN (authorization to query slack api for messages in channel) and ADMIN_IDS (authorization to use post results to the database). Create a .env file in root directory of app and create ACCESS_TOKEN and ADMIN_IDS fields in the format specified at the following link: https://www.npmjs.com/package/dotenv. Add your user_id (looks like UH1GHEGUS which you can get from your profile on slack) to ADMIN_IDS. Ping me to get the ACCESS_TOKEN.
2. yarn install
3. yarn start
4. Use postman (or similar) to make a post request to localhost:3000/postwinners with the following fields and values in body of request (make sure it is in x-www-form-urlencoded):

{
"text": "3 months",
"user_id": "[YOUR USER ID, OR WHOEVER YOU AUTHORIZED IN .ENV FILE]"
}

("3 months" is the correct answer to the poll in the #test channel in our test workspace)

5. Check database for changes to documents UH1GHEGUS and UHPKWDQJK.
